include/umf/providers/provider_ze_memory.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_ZE_MEMORY_PROVIDER_H
+#define UMF_ZE_MEMORY_PROVIDER_H
+
+#include <level_zero/ze_api.h>
+
+#include "umf/memory_provider.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// @brief ze memory provider settings struct
+typedef struct ze_memory_provider_params_t {
+    ze_context_handle_t context;
+    ze_device_handle_t device;
+    ze_device_mem_alloc_desc_t device_mem_alloc_desc;
+} ze_memory_provider_params_t;
+
+umf_memory_provider_ops_t *umfZeMemoryProviderOps(void);
```

Comment:
I think it should be called umfLevelZeroMemoryProvider (file as well). It would be more consistent with UR naming.

Reply:
changed